python/unthrow/unthrow/unthrow.pyx

```diff
@@ -0,0 +1,648 @@
+        # print("RUN_ONCE: ", mainfn, args, kwargs, self.resume_ret, self.resume_except)
+        if self.resume_stack:
+            __skip_stop=True
+            __resume_return = self.resume_ret
+            __resume_except = self.resume_except
+            # print("ABOUT TO RESUME")
+            _do_resume(<PyObject*>self.resume_stack)
+            # print("CLEANUP")
+            self.resume_stack=None
+            self.resume_except = None
+        else:
+            __skip_stop=False
+
+        self.running=1
+        interrupts_enabled=1
+        interrupt_call_level=-1
+        # we call into the first level where interrupts should be
+        interrupt_with_block_initial_level=PyEval_GetFrame().f_iblock+1
+        self.finished=True
+        # start interrupts if freq != 0
+        set_interrupt_frequency(self.freq)
+        r = None
+        err = None
+        try:
+            # print("BEG:MainFn", *args, **kwargs)
+            r = mainfn(*args, **kwargs)
+            # print("END:MainFn", *args, **kwargs)
+        except ResumableException as re:
+            # print("SUS_RET:MainFn", re.parameter)
+            r = re.parameter
+            self.resume_stack=re.saved_frames
+            re.with_traceback(None)
+            self.finished=False
+            _truncate_frame(<PyObject*>self.resume_stack)
+        except Exception as e:
+            err =  e
+        finally:
+            interrupts_enabled=0
+
+        # stop interrupts
+        global interrupts_enabled
+        set_interrupt_frequency(0)
+        self.running=0
+        interrupts_enabled=0
+        if err is not None:
+            raise err
+        return r
+
+    def set_interrupt_frequency(self,freq):
+        self.freq=freq
+        if self.running:
+            set_interrupt_frequency(self.freq)
+
+
+# like a named tuple but mutable (so that we can zero things to avoid leaks)
+class _SavedFrame:
+    def __init__(self,*,locals_and_stack,lasti,code,block_stack,globals_if_different,slow_locals):
+        dict=locals().copy() # put all the parameters to this into the dict
+        del dict["self"]
+        self._dictionary=dict
+
+    def __getattr__(self,attr):
+        if attr!="_dictionary" and attr in self._dictionary:
+            return self._dictionary[attr]
+        else:
+            super().__getattribute__(attr)
+
+    def __setattr__(self,attr,value):
+        if attr!="_dictionary" and attr in self._dictionary:
+            self._dictionary[attr]=value
+        else:
+            super().__setattr__(attr,value)
+
+    def __repr__(self):
+        return str(self._dictionary)
+
+class _PythonNULL(object):
+    pass
+
+def test_iter(maxVal):
+    for c in range(maxVal):
+        print("TI:",c)
+        yield c
+
+
+cdef int  __skip_stop=False
+
+__resume_return = None
+__resume_except = None
+
+cdef extern from "frameobject.h":
+    ctypedef struct PyTryBlock:
+        int b_type
+        int b_handler
+        int b_level
+
+
+    cdef struct _frame:
```

Comment:
This is the private cython API. But it's also used in [pybind11](https://github.com/pybind/pybind11/blob/master/include/pybind11/pybind11.h#L2641)